File: CS/Form1.cs
using DevExpress.TreeMap;
using DevExpress.XtraTreeMap;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TreemapColorizerValueProvider {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            ((TreeMapFlatDataAdapter)treeMapControl1.DataAdapter).DataSource = CreateDataSet();
            ((TreeMapColorizerBase)treeMapControl1.Colorizer).ValueProvider = new ColorizerValueProvider();
        }
        public DataTable CreateDataSet() {
            string filePath = @"..\..\Data\HPI.xml";
            if (!string.IsNullOrWhiteSpace(filePath)) {
                DataSet dataSet = new DataSet();
                dataSet.ReadXml(filePath);
                if (dataSet.Tables.Count > 0)
                    return dataSet.Tables[0];
            }
            return null;
        }
    }

    public class ColorizerValueProvider : IColorizerValueProvider {
        double IColorizerValueProvider.GetValue(IHierarchicalItem item, int itemIndex) {
            DataRowView row = item.Tag as DataRowView;
            if (row != null)
                return (double)row["HPI"];
            return 0;
        }
    }
}

File: VB/Form1.vb
Imports DevExpress.TreeMap
Imports DevExpress.XtraTreeMap
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace TreemapColorizerValueProvider
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
			CType(treeMapControl1.DataAdapter, TreeMapFlatDataAdapter).DataSource = CreateDataSet()
			CType(treeMapControl1.Colorizer, TreeMapColorizerBase).ValueProvider = New ColorizerValueProvider()
		End Sub
		Public Function CreateDataSet() As DataTable
			Dim filePath As String = "..\..\Data\HPI.xml"
			If Not String.IsNullOrWhiteSpace(filePath) Then
				Dim dataSet As New DataSet()
				dataSet.ReadXml(filePath)
				If dataSet.Tables.Count > 0 Then
					Return dataSet.Tables(0)
				End If
			End If
			Return Nothing
		End Function
	End Class

	Public Class ColorizerValueProvider
		Implements IColorizerValueProvider

		Private Function IColorizerValueProvider_GetValue(ByVal item As IHierarchicalItem, ByVal itemIndex As Integer) As Double Implements IColorizerValueProvider.GetValue
			Dim row As DataRowView = TryCast(item.Tag, DataRowView)
			If row IsNot Nothing Then
				Return DirectCast(row("HPI"), Double)
			End If
			Return 0
		End Function
	End Class
End Namespace
